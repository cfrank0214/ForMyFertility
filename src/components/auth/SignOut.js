import React, { Component } from 'react';
import {
  Container,
  Button,
  Text,
  Header,
  Content,
} from 'native-base';

import { Auth } from 'aws-amplify';

export default class SignOut extends Component {
  constructor(props) {
    super(props);
    this.signOut = this.signOut.bind(this);
  }

  signOut() {
    Auth.signOut();
    const navigation = this.props.navigation;
    navigation.navigate('Home')
  }

  render() {
    return (
      <Container>
        <Content padder >
          <Button block success onPress={() => this.signOut()}>
            <Text>
              Sign Out
          </Text>
          </Button>
        </Content>
      </Container>
    )
  }
}