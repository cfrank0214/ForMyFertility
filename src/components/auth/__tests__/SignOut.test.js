/**
 * @jest-environment jsdom
 */
import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SignOut from '../SignOut';
import {
  Container,
  Button,
  Text,
  Content,
} from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid';

describe('>>>SignOut --- Shallow Render REACT COMPONENTS', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<SignOut />)

  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <SignOut />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1)
  });
  it('The <Button /> component exists', () => {
    expect(wrapper.find(Button).exists()).toBe(true);
  });
  it('contains 1 <Button /> component', () => {
    expect(wrapper.find(Button).length).toBe(1);
  });
  it('contains 1 <Container /> component', () => {
    expect(wrapper.find(Container).length).toBe(1);
  });
  it('contains 1 <Text /> component', () => {
    expect(wrapper.find(Text).length).toBe(1);
  });
  it('contains 1 <Content /> component', () => {
    expect(wrapper.find(Content).length).toBe(1);
  });
});