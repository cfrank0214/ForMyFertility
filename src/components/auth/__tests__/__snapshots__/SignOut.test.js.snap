// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SignOut --- Shallow Render REACT COMPONENTS renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
/>
`;
